sys/include/net/netdev2_test.h

```diff
@@ -0,0 +1,306 @@
+ * #include "net/netdev2_test.h"
+ * #include "xtimer.h"
+ *
+ * #define PKT_NUMBER   (1000)
+ *
+ * static netdev2_test_t dev;
+ * static uint32_t last_start;
+ * static uint32_t sum = 0;
+ * static mutex_t wait = MUTEX_INIT;
+ *
+ * int _send_timer(netdev2_t *dev, const struct iovec *vector, int count)
+ * {
+ *     (void)dev;
+ *     (void)vector;
+ *     (void)count;
+ *
+ *     sum += (xtimer_now() - last_start);
+ *     mutex_unlock(&wait);
+ * }
+ *
+ * int main(void) {
+ *     ipv6_addr_t dst = IPV6_ADDR_UNSPECIFIED;
+ *
+ *     netdev2_test_setup(&dev, NULL);
+ *     dev->driver->init((netdev2_t *)&dev)
+ *     // initialize stack and connect `dev` to it
+ *     // ...
+ *     mutex_lock(&wait);
+ *     for (int i = 0; i < PKT_NUMBER; i++) {
+ *         last_start = xtimer_now();
+ *         conn_udp_sendto("abcd", sizeof("abcd"), NULL, 0, &dst, sizeof(dst),
+ *                         AF_INET6, 0xcafe, 0xcafe);
+ *         mutex_lock(&wait);
+ *     }
+ *     printf("Average send packet traversal time: %u\n", sum / PKT_NUMBER);
+ *     mutex_unlock(&wait);
+ *     return 0;
+ * }
+ * ~~~~~~~~~~~
+ *
+ * To provide options to the stack, the
+ * @ref netdev2_test_t::get_cbs "get callbacks" for the specific options need
+ * to be set.
+ * To catch lost packets and additional sent by the stack the send handler needs
+ * to be adapted accordingly.
+ *
+ * @{
+ *
+ * @file
+ * @brief   @ref sys_netdev2_test definitions
+ *
+ * @author  Martine Lenders <mlenders@inf.fu-berlin.de>
+ */
+#ifndef NETDEV2_TEST_H_
+#define NETDEV2_TEST_H_
+
+#include "mutex.h"
+
+#ifdef MODULE_NETDEV2_IEEE802154
+#include "net/netdev2/ieee802154.h"
+#endif
+
+#include "net/netdev2.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Callback type to handle send command
+ *
+ * @param[in] dev       network device descriptor
+ * @param[in] vector    io vector array to send
+ * @param[in] count     number of entries in vector
+ *
+ * @return  number of bytes sent
+ * @return  <= 0 on error
+ */
+typedef int (*netdev2_test_send_cb_t)(netdev2_t *dev,
+                                      const struct iovec *vector,
+                                      int count);
+
+/**
+ * @brief   Callback type to handle receive command
+ *
+ * @param[in] dev       network device descriptor
+ * @param[out] buf      buffer to write into or `NULL`
+ * @param[in] len       maximum number of bytes to read
+ * @param[out] info     status information for the received packet. Might
+ *                      be of different type for different netdev2 devices.
+ *                      May be NULL if not needed or applicable.
+ *
+ * @return <=0 on error
+ * @return number of bytes read if buf != NULL
+ * @return currently received packet size if buf == NULL
+ */
+typedef int (*netdev2_test_recv_cb_t)(netdev2_t *dev, char *buf, int len,
+                                      void *info);
+
+/**
+ * @brief   Callback type to handle device initialization.
```

Comment:
Well, it's picky but sometimes you end the line with a dot and sometimes you don't